src/common/tests_lib.c

```diff
@@ -305,3 +305,25 @@ dts_log(const char *msg, const char *file, const char *func, int line,
 	if (mask)
 		d_log(mask, "%s:%d %s() %s", file, line, func, msg);
 }
+
+void
+daos_sgl_init_with_strings(d_sg_list_t *sgl, uint32_t count, char *d, ...)
+{
+	int i;
+	va_list valist;
+	char *arg = d;
+
+	va_start(valist, d);
+
+	d_sgl_init(sgl, count);
+	for (i = 0; i < count; i++) {
+		size_t arg_len = strlen(arg) + 1;
+
+		D_ALLOC(sgl->sg_iovs[i].iov_buf, arg_len);
```

Comment:
is there a free?

Reply:
The caller will free the buffer via d_sgl_fini() finally.

Reply:
Correct, the caller will free with d_sgl_fini and must pass true for the free_iovs param. I'm changing daos_sgl_init_with_strings a little in an upcoming PR, and I'll make sure to add a comment about freeing if you're okay to land this patch now?